### ISSUE TITLE
Cursor position after C-] and line above it

### ISSUE THREAD

**Issue by liloman** [tpope/vim-surround#150]
Hi,

I find myself usually in this kind of situation (\* = cursor position):

<pre><code>
  if(something)     
      do1();*            
</code></pre>

Later I realize I needed another function inside the if body and I add:

<pre><code>
  if(something) 
      do1();
      do2();*
</code></pre>

If I select the if body in linewise mode and do:
VkS Alt-}

<pre><code>
if(something)
{*
    do1();
    do2();
}
</code></pre>

I'd rather:

<pre><code>
if(something){
    do1();
    do2();*
}
</code></pre>

But:

<pre><code>
if(something)   do1();  do2();*
</code></pre>

vBBS Alt-}

Works pretty well:

<pre><code>
if(something) {*
    do1();  do2();
}
</code></pre>

In short, I'd like:
1-Keep cursor position after Alt-} is used
2-Put { to end of the line above when Alt-} is used

Maybe, neither nice nor easy. 

Btw, great plugin!.
Edit: Fix for odd editor format...

**Comment by unphased** [comment]
Is your `Alt-}` a special bind you set up?

**Comment by liloman** [comment]
I don't think so, maybe it's just my keyboard layout (Spanish). 
I meant the } character.

Actually I'm testing this out right now and the last example ends up with:
```
if (something)   {*do1();  do2();} 
```

¿?

**Comment by unphased** [comment]
Oh ok.

So I think the wrapping with curly brackets is sort of a special case. It pretty much already puts it into K&R style (the open `{` on new line) and it's pretty quick to join the line above to get it back on to that line. 

But I think the approach for actually doing it all in one step should be to make a new custom visual-mode bind that also does the join.
